Sources/Sentry/Public/SentryDefines.h

```diff
@@ -1,5 +1,11 @@
 #import <Foundation/Foundation.h>
 
+// SentryDefines.h is a key header and will be checked early,
+// ensuring this error appears first during the compile process.
+#if APPLICATION_EXTENSION_API_ONLY_NO
+#    error Set APPLICATION_EXTENSION_API_ONLY to YES in the Sentry build settings.
+#endif
```

Comment:
`h`: I'm a bit confused. When we set the `APPLICATION_EXTENSION_API_ONLY` to yes in our Xcode project the compiler ensures we're only using APIs available to app extension. Flipping this flag for normal projects in the Sentry build settings shouldn't make a different. If we now add this check, the compilation will fail for users setting it to NO, even though it used to work for them. Or am I missing something?

Reply:
@philipphofmann, you can check the full explanation in here: https://github.com/getsentry/sentry-cocoa/pull/4599#issuecomment-2520738894 

Summary: Flipping this flag to `NO` makes our Swift internal code not available in `Sentry-Swift.h` breaking the entire project.

Reply:
Does this only happen for RN projects or also for iOS projects? I'm afraid we could break existing users setting this flag to NO on purpose.

Reply:
Any project.

> I'm afraid we could break existing users setting this flag to NO on purpose.

That's the thing. We cant set this to NO, it will break Sentry.

Reply:
Oh, that's super weird. Now, I get it thanks for explaining.